streams/src/main/java/org/apache/kafka/streams/processor/internals/FixedOrderMap.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
```

Comment:
Maybe we can put this class in `o.a.k.common.utils` instead, so it can be used elsewhere?